static/app/views/insights/common/utils/useModuleTitle.tsx

```diff
@@ -0,0 +1,17 @@
+import {DOMAIN_VIEW_MODULE_TITLES} from 'sentry/views/insights/common/utils/moduleTitles';
+import {useDomainViewFilters} from 'sentry/views/insights/pages/useFilters';
+import {MODULE_TITLES} from 'sentry/views/insights/settings';
+import type {ModuleName} from 'sentry/views/insights/types';
+
+export const useModuleTitles = (): Record<ModuleName, string> => {
+  const {isInDomainView, view} = useDomainViewFilters();
+  let moduleTitles = {...MODULE_TITLES};
+  if (isInDomainView && view) {
+    moduleTitles = {...MODULE_TITLES, ...DOMAIN_VIEW_MODULE_TITLES[view]};
+  }
+  return moduleTitles;
```

Comment:
I think this gets reconstructed quite a bit due to where it's being called, should we memoize this? Probably not really a performance issue

Reply:
Hmm, that's a good point, perhaps not a perf issue now, but maybe if we follow this pattern more it can become one. I don't think there's much harm in memoizing, so I'll do so here.

Reply:
@edwardgou-sentry, I updated to use `useMemo`!